src/audit/github_env.rs

```diff
@@ -47,8 +85,9 @@ impl WorkflowAudit for GitHubEnv {
             return Ok(findings);
         }
 
-        if let StepBody::Run { run, .. } = &step.deref().body {
-            if Self::uses_github_environment(run) {
+        if let StepBody::Run { run, shell, .. } = &step.deref().body {
+            let interpreter = shell.clone().unwrap_or("bash".into());
```

Comment:
This is unfortunately not guaranteed to be the right default, since Windows runners default to `pwsh`. I think what we need to do is look at the surrounding `NormalJob::runs_on` and maybe add some kind of `RunsOn::default_shell()` helper. I'll look into that a bit.

(Nonblocker, since this is still improving on the precision of the current approach 🙂)

Reply:
Actually, it's a bit more annoying than even that: we need to determine the step's shell via the following precedence:

1. [defaults.run.shell](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#defaultsrunshell)
2. [jobs.<job_id>.defaults.run.shell](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaultsrunshell)
3. [jobs.<job_id>.steps[*].shell](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell)
4. If none of the above are set, use the default shell implied by the runner from `runs-on` (which might be indeterminate in the self-hosted runner case)

Reply:
@woodruffw I totally missed `pwsh` on Windows as the default, I never used Windows runners and I though that `bash` was the default (via WLS). Another day, another learning I guess 🙂

Reply:
I forget about too 🙂 -- the changes in #213 should enable this, but I'll rebase off of your work so that it's not a blocker here.